src/main/java/slimeknights/tconstruct/plugin/crt/managers/AlloyingManager.java

```diff
@@ -0,0 +1,59 @@
+package slimeknights.tconstruct.plugin.crt.managers;
+
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker.api.fluid.IFluidStack;
+import com.blamejared.crafttweaker.api.item.IItemStack;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.impl.actions.recipes.ActionAddRecipe;
+import com.blamejared.crafttweaker.impl.actions.recipes.ActionRemoveRecipe;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fluids.FluidStack;
+import org.openzen.zencode.java.ZenCodeType;
+import slimeknights.mantle.recipe.FluidIngredient;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.library.recipe.alloying.AlloyRecipe;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ZenRegister
+@ZenCodeType.Name("mods.tconstruct.Allying")
+public class AlloyingManager implements IRecipeManager {
+  
+  @ZenCodeType.Method
+  public void addRecipe(String name, IFluidStack[] ingredients, IFluidStack output, int temperature) {
```

Comment:
Do you have a fluid ingredient, or just fluid stacks? It should be possible to map from that to the fluid stack, or to extend our `FluidIngredient` for that.

Reply:
We only have FluidStacks and MCtag<Fluid>.

At the end of the day, all mods are implementing their own FluidIngredient differently, so we decided that just getting a stack and passing it off to mods to turn into their own FluidIngredient would probably be the best way to do it.

Reply:
Thats probably fine, if that changes let me know and I can make sure mantle's ingredient supports it.